heat/core/dndarray.py

```diff
@@ -3348,10 +3350,11 @@ def std(self, axis=None, ddof=0, **kwargs):
         """
         return statistics.std(self, axis, ddof=ddof, **kwargs)
 
-    def __str__(self, *args):
-        # TODO: document me
-        # TODO: generate none-PyTorch str
-        return self.__array.__str__(*args)
+    def __str__(self) -> str:
+        """
+        Computes a string representation of the passed DNDarray.
+        """
+        return printing.__str__(self)
```

Comment:
Should the output resemble torch or numpy? In numpy repr() and str() are giving a slightly different output.

Reply:
Yes, I am aware of this. I was wondering also on how to proceed with this and whether to chose NumPy's behaviour (i.e. different) or Torch's (same). I think there is no discussion for ``__repr__()`` as it is supposed to 

> Return a string containing a printable representation of an object. For many types, this function makes an attempt to return a string that would yield an object with the same value when passed to ``eval()``

according to the Python documentation. For ``__str__()``, though, the Python documentation says

> ``str(object)`` returns ``object.__str__()``, which is the “informal” or nicely printable string representation of object. 

In my opinion this also entail the extra meta-information of split-axis and device as they are integral for understanding everything about the ``DNDarray's`` state.